apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ClaimResourceProviderTest.java

```diff
@@ -0,0 +1,129 @@
+package gov.cms.bfd.server.war.r4.providers;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import ca.uhn.fhir.rest.annotation.IdParam;
+import ca.uhn.fhir.rest.annotation.Read;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import com.codahale.metrics.MetricRegistry;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import org.hl7.fhir.r4.model.Claim;
+import org.hl7.fhir.r4.model.IdType;
+import org.junit.Test;
+
+public class R4ClaimResourceProviderTest {
+
+  @Test
+  public void shouldReturnCorrectIBaseResourceType() {
+    assertEquals(new R4ClaimResourceProvider().getResourceType(), Claim.class);
+  }
+
+  @Test
+  public void shouldHaveReadMethodAnnotatedWithRead() throws NoSuchMethodException {
+    R4ClaimResourceProvider provider = new R4ClaimResourceProvider();
+
+    Method readMethod = provider.getClass().getMethod("read", IdType.class, RequestDetails.class);
+
+    Read annotation = readMethod.getAnnotation(Read.class);
+
+    assertNotNull(annotation);
+  }
+
+  @Test
+  public void shouldHaveReadMethodWithAnnotatedId() throws NoSuchMethodException {
+    R4ClaimResourceProvider provider = new R4ClaimResourceProvider();
+
+    Method readMethod = provider.getClass().getMethod("read", IdType.class, RequestDetails.class);
+
+    IdParam annotation = null;
+
+    for (Parameter p : readMethod.getParameters()) {
+      if (p.getType() == IdType.class) {
+        annotation = p.getAnnotation(IdParam.class);
+      }
+    }
+
+    assertNotNull(annotation);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
Since all the checks throw the same exception checking the message here would verify the exception is actually the one the test is looking for.